agent/connect/ca/provider_vault_test.go

```diff
@@ -113,7 +113,7 @@ func TestVaultCAProvider_configureVaultAuthMethod(t *testing.T) {
 		"gcp":         {expLoginPath: "auth/gcp/login", params: map[string]interface{}{"type": "iam", "role": "test-role"}},
 		"jwt":         {expLoginPath: "auth/jwt/login"},
 		"kerberos":    {expLoginPath: "auth/kerberos/login"},
-		"kubernetes":  {expLoginPath: "auth/kubernetes/login", params: map[string]interface{}{"jwt": "fake"}},
```

Comment:
Until we officially deprecate support for passing the jwt directly, should we keep test coverage of that functionality?
Would this test case now break, since there is no role specified in the params?

Reply:
TLDR; I'm adding a legacy test case to the k8s specific tests in provider_vault_auth_test.go.

--

That test really should have failed previously as the role field is required along with the jwt. K8s requires both for login. IE. the code should have tested for the required role field but didn't. I don't check for that here as this test is more just to exercise the configureVaultAuthMethod and make sure it returns the correct auth method. I added the expanded tests for these sorts of things to the provider_vault_auth_test.go file... though looking at it I see it doesn't have a specific legacy-case check I have in a few others. I'll add that.